### PR TITLE
haproxy_cmd: relax version check 

### PR DESCRIPTION
**Pull request by dclaisse**
Check version is between a minimum and a maximum suppported version
instead of requiring a specific major version. This allows to support
more than a major version.

### CODE REVIEW

File: haproxy/haproxy_cmd/run.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func Start(sd *lib.Shutdown, cfg Config) (*dataplane.Dataplane, error) {
 		"--haproxy-bin", cfg.HAProxyPath,
 		"--config-file", cfg.HAProxyConfigPath,
 		"--reload-cmd", fmt.Sprintf("kill -SIGUSR2 %d", haCmd.Process.Pid),
+		"--restart-cmd", fmt.Sprintf("kill -SIGUSR2 %d", haCmd.Process.Pid),
```
Comment (pierrecdn) [inline haproxy/haproxy_cmd/run.go:55]: unrelated?

**Review by pierrecdn — APPROVED** [review]

File: haproxy/haproxy_cmd/run.go (hunk 1 above)
Comment (dclaisse) [reply haproxy/haproxy_cmd/run.go:55]: Indeed!
